beacon_node/beacon_chain/src/eth1_chain.rs

```diff
@@ -822,103 +702,66 @@ mod test {
-                    .eth1_data_votes
-                    .push(random_eth1_data())
-                    .expect("should push eth1 vote");
-            }
+            // Populate blocks cache with candidate eth1 blocks
+            let blocks = (start_eth1_block..end_eth1_block)
+                .map(|i| get_eth1_block(i, i))
+                .collect::<Vec<_>>();
 
-            (0..2048).for_each(|i| {
+            blocks.iter().for_each(|block| {
                 eth1_chain
                     .backend
                     .core
                     .blocks()
                     .write()
-                    .insert_root_or_child(get_eth1_block(i, i))
+                    .insert_root_or_child(block.clone())
                     .expect("should add blocks to cache");
             });
 
-            let expected_root = Hash256::from_low_u64_be(u64::max_value());
-            prev_state.eth1_data.block_hash = expected_root;
-
-            assert!(
-                prev_state.eth1_data != state.eth1_data,
-                "test requires state eth1_data are different"
-            );
-
-            store
-                .put_state(
-                    &state
-                        .get_state_root(prev_state.slot)
-                        .expect("should find state root"),
-                    prev_state,
-                )
-                .expect("should store state");
-
-            let a = eth1_chain
-                .eth1_data_for_block_production(&state, &spec)
-                .expect("should produce first random eth1 data");
-            let b = eth1_chain
+            let vote = eth1_chain
                 .eth1_data_for_block_production(&state, &spec)
-                .expect("should produce second random eth1 data");
-
-            assert!(
-                a != b,
-                "random votes should be returned if the previous eth1 data block hash is unknown"
-            );
-        }
-    }
-
-    mod prev_block_hash {
-        use super::*;
-        use store::MemoryStore;
-
-        #[test]
-        fn without_store_lookup() {
-            let spec = &E::default_spec();
-            let store = Arc::new(MemoryStore::open());
-
-            let state: BeaconState<E> = BeaconState::new(0, get_eth1_data(0), spec);
+                .expect("should produce default eth1 data vote");
 
             assert_eq!(
-                eth1_block_hash_at_start_of_voting_period(store, &state),
-                Ok(state.eth1_data.block_hash),
-                "should return the states eth1 data in the first half of the period"
+                vote,
+                blocks
+                    .last()
+                    .expect("should have blocks")
+                    .clone()
+                    .eth1_data()
+                    .expect("should have valid eth1 data"),
+                "default vote must correspond to last block in candidate blocks"
             );
         }
 
+        /* FIXME(sproul): delete? annoying
```

Comment:
Thoughts on deleting this test? It seems to be testing `eth1_block_hash_at_start_of_voting_period` which no longer exists, but I haven't thought if there's a good replacement. CC @pawanjay176 @paulhauner

Reply:
I couldn't think of an equivalent test in the new voting scheme either. The closest I could think of is the `eth1_data_empty_cache` test where we vote for `state.eth1_data`.

Reply:
Yep delete!